integration/test/organization/main_test.go

```diff
@@ -0,0 +1,137 @@
+// +build k8srequired
+
+package organization
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/core/v1alpha1"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/v2/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/crd"
+	"github.com/giantswarm/apptest"
+	"github.com/giantswarm/micrologger"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-admission-controller/integration/env"
+)
+
+const (
+	testCatalogName = "control-plane-test"
+	prodCatalogName = "control-plane"
+	testCatalogUrl  = "https://giantswarm.github.io/control-plane-test-catalog"
+	prodCatalogUrl  = "https://giantswarm.github.io/control-plane-catalog"
+	// API Groups for upstream Cluster API types.
+	clusterAPIGroup                    = "cluster.x-k8s.io"
+	infrastructureAPIGroup             = "infrastructure.cluster.x-k8s.io"
+	experimentalClusterAPIGroup        = "exp.cluster.x-k8s.io"
+	experimentalInfrastructureAPIGroup = "exp.infrastructure.cluster.x-k8s.io"
+)
+
+var (
+	appTest apptest.Interface
```

Comment:
Do we need to make this instance global variable? Is there something in the `apptest` design or implementation that requires it or makes it convenient to have it that way? I noticed that tests are using shared client from that instance and it brings high risk of subtle bugs in tests due to overlapping CRs. When dealing with CRDs it quickly becomes tedious and we often make helper functions for that, but this often then leads to shared CR names / IDs because normally those don't matter but in this specific case it's suddenly very important aspect.

Reply:
The apptest library is still pretty fresh. One of the goals is to have simple wiring in components like metrics-server where we only use Go for the integration tests.

The clients are a convenience wrapper for that. But especially with the controller client I can see how it could cause problems. They don't have to be used and could even be removed. All that is required is a kubeconfig.

Reply:
We talked with Jose about this and when the test is performed towards shared k8s API due to nature of admission controller, the shared client can't be avoided in the end. We just need to be careful when writing tests that we don't share IDs between them :slightly_smiling_face:  All good here. :+1: